src/main/java/RedisSessionManagerT8.java

```diff
@@ -0,0 +1,309 @@
+            log.error("Error - Context: createSession. Description: " + e.getMessage());
+        }
+
+        return session;
+    }
+
+    /**
+     * Add session to this manager
+     * @param session
+     */
+    @Override
+    public void add(Session session) {
+        try {
+            saveSession(session);
+        } catch (IOException e) {
+            log.error("Error - Context: add. Description: " + e.getMessage());
+        }
+    }
+
+
+    @Override
+    public Session findSession(String id) throws IOException {
+        try {
+
+            Hashtable<String, byte[]> result = withRedis(
+                    (Jedis jedis)-> {
+                        Hashtable<String, byte[]> r =  new Hashtable<>();
+                        r.put(REDIS_METADATA_KEY, jedis.get((id + REDIS_METADATA_KEY).getBytes()));
+                        r.put(REDIS_ATTRIBUTES_KEY, jedis.get((id + REDIS_ATTRIBUTES_KEY).getBytes()));
+                        return r;
+                    }
+            );
+
+            Hashtable<String, Object> metadata =
+                    (Hashtable)SerializationUtils.deserialize(Base64.getDecoder().decode(result.get(REDIS_METADATA_KEY)));
+
+            Hashtable<String, Object> attributes =
+                    (Hashtable)SerializationUtils.deserialize(Base64.getDecoder().decode(result.get(REDIS_ATTRIBUTES_KEY)));
+
+            return getSession(metadata, attributes);
+
+        } catch (Exception e) {
+            log.error("Error - Context: findSession. Description: " + e.getMessage());
+            throw new IOException("Session not found with id: " + id);
+        }
+    }
+
+    @Override
+    public void remove(Session session) {
+
+        remove(session, false);
+    }
+
+    @Override
+    public void remove(Session session, boolean update) {
+        //Redis will remove the attributes
+    }
+
+    @Override
+    public void processExpires() {
+        // We are going to use Redis's ability to expire keys for session expiration.
+        // Do nothing.
+    }
+
+    //------------------------------------------------------------------------------------------------------------------
+    // END Sessions
+    //------------------------------------------------------------------------------------------------------------------
+
+
+    //------------------------------------------------------------------------------------------------------------------
+    // START Utils
+    //------------------------------------------------------------------------------------------------------------------
+
+    /**
+     * Get session from metadata and attributes
+     * @param metadata
+     * @param attributes
+     * @return
+     */
+    private RedisSessionT8 getSession(Hashtable<String, Object> metadata, Hashtable<String, Object> attributes){
+        RedisSessionT8 session = (RedisSessionT8) createEmptySession();
+        session.setValid(Boolean.valueOf((String) metadata.get(METADATA_VALID)));
```

Comment:
You need to add significant error handling here. You're assuming a lot about the state of `metadata` and its value for `METADATA_VALID`.